### DISCUSSION TITLE
Estimated release date for react Spectrum 2

### DISCUSSION THREAD

**Discussion by cheema-corellian** [adobe/react-spectrum#6663] [Q&A] [answered]
We currently use Spectrum React for our web application.  We sometimes run into a limitation with it and think that perhaps Spectrum 2 will offer better options.  Spectrum 2 is scheduled to be released sometime in 2024, per https://s2.spectrum.adobe.com/.  Does anybody have a more precise estimate?

Also, is it being developed in collaboration with the community with changes being checked into GitHub somewhere?  I wouldn't mind checking it out before official release, if possible.

**Comment by snowystinger** [comment, marked as answer]
We do not currently have a release date, the link you posted is the best estimate at the moment. We should have more updates as we progress through the year. Thank you for your patience.

**Comment by leonsbuddydave** [comment]
Hi, is there any update to this release timeline?

**Reply by devongovett** [reply]
It is released: http://react-spectrum.adobe.com/s2/. We are working on a full documentation website, but the storybook serves as documentation for now.

**Reply by cheema-corellian** [reply]
@devongovett the storybook site doesn't show some critical components. e.g. DateField and DatePicker etc. Are these missing components not migrated yet or not documented on Storybook yet?

**Reply by devongovett** [reply]
they were added recently (https://github.com/adobe/react-spectrum/pull/8428) and will be in the next release.

**Reply by cheema-corellian** [reply]
That is great to hear. Thank you!

Is it correct to assume that after the addition of these components, S2 will have feature parity with S1? Or are there other components or functions that still need to be migrated? I ask because I would like to start the work to migrate our existing codebase over to S2. But may wait if S2 is still missing some functionality that is present in S1.

**Reply by leonsbuddydave** [reply]
@devongovett Thank you!

**Reply by snowystinger** [reply]
> S2 will have feature parity with S1

Not quite yet but we are getting very close. You'd need to check based on the functionality you need, it may be that we have everything you use already.

Make sure to check out the migration guide to help you as well https://react-spectrum.adobe.com/s2/index.html?path=/docs/migrating--docs
